src/Raven.Studio/typescript/viewmodels/database/indexes/additionalSourceSyntax.ts

```diff
@@ -1,29 +1,46 @@
 import dialogViewModelBase = require("viewmodels/dialogViewModelBase");
+import copyToClipboard = require("common/copyToClipboard");
 
 class additionalSourceSyntax extends dialogViewModelBase {
 
+    htmlElement: HTMLElement;
 
-    static readonly additionalSourceCsharp = `// *Additional Source*
+    compositionComplete() {
+        super.compositionComplete();
+        this.bindToCurrentInstance("copySample");
+        this.htmlElement = document.getElementById("additionalSourceSyntaxDialog");
+    }
+
+    copySample(sampleTitle: string) {
+        let sampleArray = sampleTitle.startsWith("Csharp") ? additionalSourceSyntax.csharpSamples : additionalSourceSyntax.javascriptSamples;
```

Comment:
can't we loop in both array = then no need to relay on starts with + use const not let.

Reply:
done